src/main/java/com/redhat/devtools/lsp4ij/features/documentSymbol/LSPDocumentSymbolSupport.java

```diff
@@ -0,0 +1,101 @@
+package com.redhat.devtools.lsp4ij.features.documentSymbol;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiFile;
+import com.redhat.devtools.lsp4ij.LSPRequestConstants;
+import com.redhat.devtools.lsp4ij.LanguageServerItem;
+import com.redhat.devtools.lsp4ij.LanguageServiceAccessor;
+import com.redhat.devtools.lsp4ij.features.AbstractLSPDocumentFeatureSupport;
+import com.redhat.devtools.lsp4ij.internal.CancellationSupport;
+import com.redhat.devtools.lsp4ij.internal.CompletableFutures;
+import org.eclipse.lsp4j.DocumentSymbol;
+import org.eclipse.lsp4j.DocumentSymbolParams;
+import org.eclipse.lsp4j.SymbolInformation;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * LSP document symbol support which loads and caches symbol result by consuming:
+ *
+ * <ul>
+ *     <li>LSP 'textDocument/documentSymbol' requests</li>
+ * </ul>
+ */
+public class LSPDocumentSymbolSupport extends AbstractLSPDocumentFeatureSupport<DocumentSymbolParams, List<DocumentSymbolData>> {
+    public LSPDocumentSymbolSupport(@NotNull PsiFile file) {
+        super(file);
+    }
+
+    public CompletableFuture<List<DocumentSymbolData>> getDocumentSymbols(DocumentSymbolParams params) {
+        return super.getFeatureData(params);
+    }
+
+    @Override
+    protected CompletableFuture<List<DocumentSymbolData>> doLoad(DocumentSymbolParams documentSymbolParams, CancellationSupport cancellationSupport) {
+        PsiFile file = super.getFile();
+        return getDocumentSymbols(file.getVirtualFile(), file.getProject(), documentSymbolParams, cancellationSupport);
+    }
+
+    private static @NotNull CompletableFuture<List<DocumentSymbolData>> getDocumentSymbols(@NotNull VirtualFile file,
+                                                                                           @NotNull Project project,
+                                                                                           @NotNull DocumentSymbolParams params,
+                                                                                           @NotNull CancellationSupport cancellationSupport) {
+
+        return LanguageServiceAccessor.getInstance(project)
+                .getLanguageServers(file, LanguageServerItem::isDocumentLinkSupported)
```

Comment:
You need to create a new method `LanguageServerItem:#isDocumentSymbolSupported` to check the proper server capability